core/controlplane/config/templates/cloud-config-worker

```diff
@@ -416,14 +428,15 @@ ssh_authorized_keys:
       content: |
         [Unit]
         Description=Pull and tag a mirror image for pause-amd64
-        After=docker.service
         Wants=docker.service
 
         [Service]
-        Restart=on-failure
+        Type=oneshot
         RemainAfterExit=true
         ExecStartPre=/usr/bin/docker pull {{.PauseImage.RepoWithTag}}
```

Comment:
You probably want to some retry logic to make this reliable against e.g. a temporary network failure. AFAIK, `oneshot` services don't restart on failures.